exporter/awsxrayexporter: Fix bug with status code

xibz opened the pull request:
This fixes a bug to ensure ensures that if the status code is a string,
that it'll parse it as an int and properly set the status code.

Fixes #6662

Signed-off-by: benjamin_j_powell <benjamin_j_powell@apple.com>

bogdandrutu commented:
Please sign the CLA, for re-testing the CLA you can simply just comment "/easycla"

xibz committed:
exporter/awsxrayexporter: Fix bug with status code

This fixes a bug to ensure ensures that if the status code is a string,
that it'll parse it as an int and properly set the status code.

Signed-off-by: xibz <impactbchang@gmail.com>

xibz commented:
@bogdandrutu looks like it figured itself out :p

bogdandrutu commented on exporter/awsxrayexporter/internal/translator/http.go:
```diff
@@ -54,7 +54,14 @@ func makeHTTP(span pdata.Span) (map[string]pdata.AttributeValue, *awsxray.HTTPDa
+			statusCode := value.IntVal()
+			if value.Type() == pdata.AttributeValueTypeString {
+				statusCodeStr := value.StringVal()
+				v, _ := strconv.ParseInt(statusCodeStr, 10, 64)
+				statusCode = v
```
Why ignoring the error?

xibz replied:
Originally I was logging it, but I noticed a few errors where errors were ignored, and since there is no logging in this package, I decided to ignore it. One option is to log it using `zap.L()` but nothing sets loggers up on the global space nor are they really used anywhere except the processors (and I think a couple other areas). So because of this, I thought ignoring may be the best option. However, if we want to log it using `zap.L()` it'll be a noop. Another option is to pass a logger in, but that would require passing it all the way down, which the exporters do not get. So that would need to change.

bogdandrutu replied:
We definitely pass a Logger to all the components (receivers, processors, exporters), see https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/receiver.go#L120 (equivalent for all, we send a create settings which includes telemetry settings), see https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/telemetry.go#L23.

xibz replied:
Ah! Don't know how I missed the loggers in the exporters. I swear I tried grepping a little bit ago, but they are there. Let me log that error.

xibz replied:
@bogdandrutu Alright, updated and no I am passing the logger down to the internal package and logging the error. Please take a look when you get a chance!

xibz replied:
So there looks to be already a parsing helper that can be set in the config. I'll go ahead and set that, and things should be parsed correctly
